File: algos/binary_search.js
/*
* @param {number[]} array
* @param {number} target
* @return {bool}
*/
var binary_search = function(array, target) {
    start = 0;
    end = array.length;
    while (start < end) {
        middle = Math.floor(start + (end - start)/2)
        if (array[middle] === target) { return true }

        if (array[middle] < target) { 
            start = middle + 1
        } else {
            end = middle
        }
    }
    return false
}

console.log(binary_search([1, 3, 11, 12, 14, 18, 18, 19], 19))

File: algorithms/binary_search.py
def binary_search(input_list: list, target: int) -> bool:
	start: int = 0
	end: int = len(input_list)
	while (end > start):
		middle: int = int((end + start)/2)

		if target == input_list[middle]:
			return True
		elif target < input_list[middle]:
			end = middle
		else:
			start = middle + 1
	return False

input_list = [1, 3, 11, 12, 14, 18, 18, 19]
print()
print(binary_search(input_list, 18))

